File: slp/backend-dotnet/Features/Session/SessionTokenService.cs
using System.Security.Cryptography;
namespace backend_dotnet.Features.Session
{
    public static class SessionTokenService
    {
        public static string GenerateToken()
        {
            var bytes = RandomNumberGenerator.GetBytes(32);
            return Convert.ToBase64String(bytes);
        }

        public static string HashToken(string token)
        {
            using var sha = SHA256.Create();
            var bytes = sha.ComputeHash(System.Text.Encoding.UTF8.GetBytes(token));
            return Convert.ToBase64String(bytes);
        }
    }
}

File: slp/backend-java/src/main/java/com/app/backendjava/features/session/SessionTokenService.java
package com.app.backendjava.features.session;

import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;
import java.security.SecureRandom;
import java.util.Base64;
import java.nio.charset.StandardCharsets;

public class SessionTokenService {

    private static final SecureRandom secureRandom = new SecureRandom();

    /**
     * Generates a random 32-byte token and returns it as a Base64 string.
     */
    public static String generateToken() {
        byte[] bytes = new byte[32];
        secureRandom.nextBytes(bytes);
        return Base64.getEncoder().encodeToString(bytes);
    }

    /**
     * Hashes a token string using SHA-256 and returns the Base64 result.
     */
    public static String hashToken(String token) {
        try {
            MessageDigest digest = MessageDigest.getInstance("SHA-256");
            byte[] hash = digest.digest(token.getBytes(StandardCharsets.UTF_8));
            return Base64.getEncoder().encodeToString(hash);
        } catch (NoSuchAlgorithmException e) {
            throw new RuntimeException("Error hashing session token", e);
        }
    }
}